lmdeploy/turbomind/deploy/converter.py

```diff
@@ -134,6 +135,8 @@ def get_output_model_registered_name_and_config(model_path: str,
     weight_type = 'fp16'
 
     config = TurbomindModelConfig.from_dict({}, allow_none=True)
+    for field in dataclasses.fields(TurbomindModelConfig):
```

Comment:
I think we'd better change `TurbomindModelConfig.update(self, other: 'TurbomindModelConfig') to `TurbomindModelConfig.update_from_engine_config(self, other: 'TurbomindEngineConfig')
@irexyc

Reply:
Both [cfg](https://github.com/InternLM/lmdeploy/blob/bdf2bba4a733cb6d3d9b06ed2f8ef68a3b2b97a8/lmdeploy/turbomind/turbomind.py#L248) and [engine_config](https://github.com/InternLM/lmdeploy/blob/bdf2bba4a733cb6d3d9b06ed2f8ef68a3b2b97a8/lmdeploy/turbomind/turbomind.py#L252) have `session_len` and we should use the value in engine_config.

Maybe we could update cfg.session_len if engine_config.session_len is not None before `self.config.update(cfg)`

Reply:
@QwertyJack Can I commit my modification to your PR?

Reply:
Of course.

Reply:
I opened a PR https://github.com/QwertyJack/lmdeploy/pull/1 to your repo.
May review it.

Reply:
Done